cms/djangoapps/contentstore/views/course_optimizer.py

```diff
@@ -0,0 +1,218 @@
+"""
+These views handle all actions in Studio related to link checking of
+courses
+"""
+
+
+import json
+import logging
+import os
+from wsgiref.util import FileWrapper
+
+from django.conf import settings
+from django.contrib.auth.decorators import login_required
+from django.core.exceptions import PermissionDenied
+from django.db import transaction
+from django.http import Http404, HttpResponse, StreamingHttpResponse
+from django.utils.translation import gettext as _
+from django.views.decorators.csrf import ensure_csrf_cookie
+from django.views.decorators.http import require_GET, require_http_methods
+from opaque_keys.edx.keys import CourseKey
+from user_tasks.conf import settings as user_tasks_settings
+from user_tasks.models import UserTaskArtifact, UserTaskStatus
+
+from common.djangoapps.student.auth import has_course_author_access
+from common.djangoapps.util.json_request import JsonResponse
+from common.djangoapps.util.views import ensure_valid_course_key
+from cms.djangoapps.contentstore.xblock_storage_handlers.view_handlers import get_xblock
+from cms.djangoapps.contentstore.xblock_storage_handlers.xblock_helpers import usage_key_with_run
+from xmodule.modulestore.django import modulestore  # lint-amnesty, pylint: disable=wrong-import-order
+
+from ..tasks import CourseLinkCheckTask, check_broken_links
+from ..utils import reverse_course_url
+
+__all__ = [
+    'link_check_handler',
+    'link_check_status_handler',
+]
+
+log = logging.getLogger(__name__)
+
+STATUS_FILTERS = user_tasks_settings.USER_TASKS_STATUS_FILTERS
```

Comment:
A one line comment as to how these will be used would be helpful. The module filters user tasks on the basis of these, but I'm unfamiliar with this package's use. The one line saves me the trouble of having to go read up on the package if all I want is some notion of what it's used for.

Reply:
The comment above "# Tuple containing zero or more filters for UserTaskStatus listing REST API calls." does not help my understanding at all. I understand we apply filters. What do we apply them for?

I went through code quite a bit to figure that out, which was not straightforward.

The result:
This is an adjustable django setting. It specifies general filters that are applied to all tasks before they are returned to the user. Currently this setting is not specified in edx-app settings. So it defaults to a default filter. Which is quite hard to find to figure out where this is defined (in some other repo called `django_user_tasks`.
This is the code place:
https://github.com/openedx/django-user-tasks/blob/20e4e6eb81f5e981d5bbdba390ffcf8e6c3e0d0a/user_tasks/filters.py#L9

So it will be quite hard for a user to understand what the filter actually does or what it's purpose is.

The explaining comments for the default filter are:
```
    Default filter for UserTaskArtifact listings in the REST API.

    Ensures that superusers can see all artifacts, but other users
    can only see artifacts for tasks they personally triggered.
```
And then more specific:
```
Filter out any artifacts which the requesting user does not have permission to view.
```

On top of that, there are two sets of these filters - one is for task artifacts, one for statuses. They function the exact same way.

This is important to capture on edx-platform in some way. Maybe something like: "These filters can be overridden in django settings of edx-platform. If they are not, the default behavior is the following: ..."